modules/core/src/main/scala-3/sangria/macros/ToExprGivens.scala

```diff
@@ -0,0 +1,414 @@
+            ${ Expr(t) },
+            ${ Expr(d) },
+            ${ Expr(dirs) },
+            ${ Expr(c) },
+            ${ Expr(p) })
+        }
+
+  given ToExpr[InputValueDefinition] with
+    def apply(inpValDef: InputValueDefinition)(using Quotes) = inpValDef match
+      case InputValueDefinition(n, v, de, di, desc, c, p) =>
+        '{
+          sangria.ast.InputValueDefinition(
+            ${ Expr(n) },
+            ${ Expr(v) },
+            ${ Expr(de) },
+            ${ Expr(di) },
+            ${ Expr(desc) },
+            ${ Expr(c) },
+            ${ Expr(p) })
+        }
+
+  given ToExpr[OperationTypeDefinition] with
+    def apply(inpOpTpeDef: OperationTypeDefinition)(using Quotes) = inpOpTpeDef match
+      case OperationTypeDefinition(o, t, c, p) =>
+        '{ OperationTypeDefinition(${ Expr(o) }, ${ Expr(t) }, ${ Expr(c) }, ${ Expr(p) }) }
+
+  given ToExpr[EnumValueDefinition] with
+    def apply(enumValDef: EnumValueDefinition)(using Quotes) = enumValDef match
+      case EnumValueDefinition(n, d, desc, c, p) =>
+        '{
+          EnumValueDefinition(
+            ${ Expr(n) },
+            ${ Expr(d) },
+            ${ Expr(desc) },
+            ${ Expr(c) },
+            ${ Expr(p) })
+        }
+
+  given ToExpr[FieldDefinition] with
+    def apply(liftFieldDef: FieldDefinition)(using Quotes) = liftFieldDef match
+      case FieldDefinition(n, f, a, d, desc, c, p) =>
+        '{
+          FieldDefinition(
+            ${ Expr(n) },
+            ${ Expr(f) },
+            ${ Expr(a) },
+            ${ Expr(d) },
+            ${ Expr(desc) },
+            ${ Expr(c) },
+            ${ Expr(p) })
+        }
+
+  given ToExpr[DirectiveLocation] with
+    def apply(dirLocDef: DirectiveLocation)(using Quotes) = dirLocDef match
+      case DirectiveLocation(n, c, p) =>
+        '{ DirectiveLocation(${ Expr(n) }, ${ Expr(c) }, ${ Expr(p) }) }
+
+  given ToExpr[Definition] with
+    def apply(definition: Definition)(using Quotes) = definition match
+      case OperationDefinition(o, n, v, d, s, c, tc, p) =>
+        '{
+          sangria.ast.OperationDefinition(
+            ${ Expr(o) },
+            ${ Expr(n) },
+            ${ Expr(v) },
+            ${ Expr(d) },
+            ${ Expr(s) },
+            ${ Expr(c) },
+            ${ Expr(tc) },
+            ${ Expr(p) })
+        }
+      case FragmentDefinition(n, t, d, s, v, c, tc, p) =>
+        '{
+          sangria.ast.FragmentDefinition(
+            ${ Expr(n) },
+            ${ Expr(t) },
+            ${ Expr(d) },
+            ${ Expr(s) },
+            ${ Expr(v) },
+            ${ Expr(c) },
+            ${ Expr(tc) },
+            ${ Expr(p) })
+        }
+
+      case DirectiveDefinition(n, a, l, desc, c, p) =>
+        '{
+          sangria.ast.DirectiveDefinition(
+            ${ Expr(n) },
+            ${ Expr(a) },
+            ${ Expr(l) },
+            ${ Expr(desc) },
+            ${ Expr(c) },
+            ${ Expr(p) })
+        }
```

Comment:
One field was added in the `DirectiveDefinition` in https://github.com/sangria-graphql/sangria/pull/899 that should be backported here to fix the compilation error:
```
[error] 128 |      case DirectiveDefinition(n, a, l, desc, c, p) =>
[error]     |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[error]     |Wrong number of argument patterns for sangria.ast.DirectiveDefinition; expected: (String, Vector[sangria.ast.InputValueDefinition], Vector[sangria.ast.DirectiveLocation], Option[sangria.ast.StringValue], Boolean, Vector[sangria.ast.Comment], Option[sangria.ast.AstLocation])
```

```suggestion
      case DirectiveDefinition(n, a, l, desc, r, c, p) =>
        '{
          sangria.ast.DirectiveDefinition(
            ${ Expr(n) },
            ${ Expr(a) },
            ${ Expr(l) },
            ${ Expr(desc) },
            ${ Expr(r) },
            ${ Expr(c) },
            ${ Expr(p) })
        }
```